src/frontend-samples/explode-sample/ExplodeTile.ts

```diff
@@ -0,0 +1,629 @@
+      range: parent.range,
+      maximumSize: parent.maximumSize,
+    }, parent.tree);
+
+    this.setExplodeTransform(ExplodeApp.explodeAttributes.min);
+  }
+
+  public computeLoadPriority(_viewports: Iterable<Viewport>): number {
+    // We want the element's graphics to be updated as soon as possible
+    return 0;
+  }
+
+  /** Creates an unique id for requesting tiles from the backend. */
+  private makeRequestId(): string {
+    const requestId = requestIdSequence.next();
+    if (requestId.done)
+      return (-1).toString(16);
+    return requestId.value.toString(16);
+  }
+
+  /** No children tiles. */
+  protected _loadChildren(resolve: (children: Tile[] | undefined) => void, _reject: (error: Error) => void): void {
+    resolve(undefined);
+  }
+
+  /** Return a Promise that resolves to the raw data representing this tile's content. */
+  public async requestContent(_isCanceled: () => boolean): Promise<TileRequest.Response> {
+    const props: ElementGraphicsRequestProps = {
+      id: this.makeRequestId(),
+      elementId: this.data.elementId,
+      toleranceLog10: this.toleranceLog10,
+      formatVersion: this.formatVersion,
+      clipToProjectExtents: false,
+    };
+    return IModelApp.tileAdmin.requestElementGraphics(this.tree.iModel, props);
+  }
+
+  /** Return a Promise that deserializes this tile's content from raw format produced by [[requestContent]]. */
+  public async readContent(data: TileRequest.ResponseData, system: RenderSystem, isCanceled?: () => boolean): Promise<TileContent> {
+    if (undefined === isCanceled)
+      isCanceled = () => !this.isLoading;
+
+    assert(data instanceof Uint8Array);
+    const stream = new ByteStream(data.buffer);
+
+    const position = stream.curPos;
+    const format = stream.nextUint32;
+    stream.curPos = position;
+
+    assert(TileFormat.IModel === format);
+
+    const tree = this.tree;
+    const reader = ImdlReader.create(stream, tree.iModel, tree.modelId, tree.is3d, system, undefined, undefined, isCanceled, undefined, this.contentId);
+
+    let content: TileContent = { isLeaf: true };
+    if (reader) {
+      try {
+        content = await reader.read();
+      } catch (_) {
+        //
+      }
+    }
+
+    (this.tree as ExplodeTileTree).elementContentLoaded(this.data.elementId);
+    return content;
+  }
+
+  /** Updates the content range and transform applied to the graphics with a changed explode scaling. */
+  public setExplodeTransform(explodeScaling: number) {
+    if (explodeScaling === this._prevExplodeFactor) return;
+    this._explodeTransform = calculateExplodeTransform(this.data.boundingBox.center, this.centerOfMass, explodeScaling);
+    this._explodeTransform.multiplyRange(this.data.boundingBox, this._contentRange);
+  }
+
+  /** Output this tile's graphics. */
+  public drawGraphics(args: ExplodeTileDrawArgs) {
+    const gfx = this.produceGraphics();
+    if (!gfx) return;
+    this.setExplodeTransform(args.explodeFactor);
```

Comment:
This isn't safe - you are not guaranteed that `args` is an instance of your subclass. Can you store the explodeFactor on the TileTree instead?